server/src/main/java/org/elasticsearch/indices/IndicesWriteLoadStore.java

```diff
@@ -0,0 +1,201 @@
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class IndicesWriteLoadStore implements Closeable {
+    public static final Setting<Boolean> ENABLED_SETTING = Setting.boolSetting(
+        "indices.write_load.store.enabled",
+        true,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+    public static final Setting<Integer> MAX_RETRIES_SETTING = Setting.intSetting(
+        "indices.write_load.store.max_retries",
+        3,
+        1,
+        Setting.Property.NodeScope
+    );
+    // Average document size is ~450 bytes -> 1Mb / 450b = ~2000 docs
+    public static final Setting<ByteSizeValue> MAX_BULK_SIZE_SETTING = Setting.byteSizeSetting(
+        "indices.write_load.store.max_bulk_size",
+        ByteSizeValue.ofMb(1),
+        Setting.Property.NodeScope
+    );
+    public static final Setting<Integer> MAX_DOCUMENTS_PER_BULK_SETTING = Setting.intSetting(
+        "indices.write_load.store.max_documents",
+        2000,
+        1,
+        10000, // That sets an upper bound of ~5Mb
+        Setting.Property.NodeScope
+    );
+    public static final Setting<Integer> MAX_CONCURRENT_REQUESTS_SETTING = Setting.intSetting(
+        "indices.write_load.store.max_concurrent_requests",
+        1,
+        0,
+        10,
+        Setting.Property.NodeScope
+    );
+
+    public static final String INDICES_WRITE_LOAD_DATA_STREAM = ".indices-write-load-distribution";
+
+    public static final String INDICES_WRITE_LOAD_FEATURE_NAME = "indices_write_load";
+
+    public static final SystemDataStreamDescriptor INDICES_WRITE_LOAD_DATA_STREAM_DESCRIPTOR = loadDataStreamDescriptor();
+
+    private static SystemDataStreamDescriptor loadDataStreamDescriptor() {
+        try (
+            XContentParser parser = XContentType.JSON.xContent()
+                .createParser(XContentParserConfiguration.EMPTY, loadComposableIndexTemplate())
+        ) {
+            ComposableIndexTemplate composableIndexTemplate = ComposableIndexTemplate.parse(parser);
+            return new SystemDataStreamDescriptor(
+                INDICES_WRITE_LOAD_DATA_STREAM,
+                "Stores indices write load distribution over time",
+                SystemDataStreamDescriptor.Type.INTERNAL,
+                composableIndexTemplate,
+                Map.of(),
+                Collections.emptyList(),
+                ExecutorNames.DEFAULT_SYSTEM_DATA_STREAM_THREAD_POOLS
```

Comment:
I wonder if we should use the `DEFAULT_SYSTEM_INDEX_THREAD_POOLS` instead to reduce interference from user-invoked indexing?